syntax = "proto3";

package NameNode.Service;

message UploadDBRequest {
    bytes db = 1;
}

message UploadDBResponse {
    string status = 1;
    string message = 2;
}

message HeartBeatRequest {
    string name = 1;
    string ip = 2;
    int32 port = 3;
}

message HeartBeatResponse {
    string status = 1;
    string message = 2;
}

message DownloadDBRequest {
    string name = 1;
}

message DownloadDBResponse {
    bool canDownload = 1;
    bytes db = 2;
}

service DBService {
    rpc UploadDB(UploadDBRequest) returns (UploadDBResponse) {}
    rpc DownloadDB(DownloadDBRequest) returns (DownloadDBResponse) {}
    rpc HeartBeat(HeartBeatRequest) returns (HeartBeatResponse) {}
}
